consensus/module_consensus_state_sync.go

```diff
@@ -22,13 +23,36 @@ func (m *consensusModule) GetNodeAddress() string {
 	return m.nodeAddress
 }
 
-// TODO(#352): Implement this function, currently a placeholder.
-// commitReceivedBlocks commits the blocks received from the blocksReceived channel
+// blockApplicationLoop commits the blocks received from the blocksReceived channel
 // it is intended to be run as a background process
 func (m *consensusModule) blockApplicationLoop() {
-	// runs as a background process in consensus module
-	// listens on the blocksReceived channel
-	// commits the received block
+	for blockResponse := range m.blocksReceived {
+		block := blockResponse.Block
+		maxPersistedHeight, err := m.maxPersistedBlockHeight()
+		if err != nil {
+			m.logger.Err(err).Msg("couldn't query max persisted height")
+			return
+		}
+
+		if block.BlockHeader.Height <= maxPersistedHeight || block.BlockHeader.Height > m.CurrentHeight() {
```

Comment:
Can you split this into 2 different if structures with specialized info messages?